OpenRA.Game/Graphics/WorldRenderer.cs

```diff
@@ -109,8 +109,8 @@ public void ReplacePalette(string name, IPalette pal)
 			palette.ReplacePalette(name, pal);
 
 			// Update cached PlayerReference if one exists
-			if (palettes.ContainsKey(name))
-				palettes[name].Palette = pal;
+			if (palettes.TryGetValue(name, out var paletteReference))
+				paletteReference.Palette = pal;
```

Comment:
The above file, `HardwarePalette.cs`, has an exception to the rule because "False positive - indexer is a set not a get."
Is this not the same? Am I missing something?

Reply:
You can assign to `paletteReference.Palette` but not to
```
if (palettes.TryGetValue(name, out var palette))
	palette = new ImmutablePalette(p);
```
(the latter won't do because it only overwrites the local variable.)